_dev/src/ts/modals/StartUpdateModal.ts

```diff
@@ -0,0 +1,70 @@
+import ComponentAbstract from '../types/ComponentAbstract';
+import api from '../api/RequestHandler';
+
+export default class StartUpdateModal extends ComponentAbstract {
+  protected readonly formId = 'form-confirm-update';
+  protected readonly confirmCheckboxId = 'modal-start-update-own-backup';
+
+  public mount() {
+    this.form.addEventListener('submit', this.onSubmit);
+    this.form.addEventListener('change', this.onChange);
+
+    this.updateSubmitButtonStatus(
+      document.getElementById('modal-start-update-own-backup') as HTMLInputElement | undefined
+    );
+  }
+
+  public beforeDestroy() {
+    this.form.removeEventListener('submit', this.onSubmit);
+    this.form.removeEventListener('change', this.onChange);
+  }
+
+  private get form(): HTMLFormElement {
+    const form = document.forms.namedItem('form-confirm-update');
+    if (!form) {
+      throw new Error('Form not found');
+    }
+
+    ['routeToSubmit'].forEach((data) => {
+      if (!form.dataset[data]) {
+        throw new Error(`Missing data ${data} from form dataset.`);
+      }
+    });
```

Comment:
```suggestion
   if (!form.dataset.routeToSubmit) {
    throw new Error('Missing data routeToSubmit from form dataset.');
}
```

If the intention is to make this extensible, it would be wise to clarify that intention in a comment or use a structure that explicitly calls for generalization.

Reply:
I've wrote a comment a few lines above